### DISCUSSION TITLE
Solana Pay: reference not included in transaction

### DISCUSSION THREAD

**Discussion by matbech** [phantom/docs#190] [General]
The reference parameter in the Solana Pay url is not included in a token transfer transaction. Per solana pay specification:
https://docs.solanapay.com/spec
"If the values are provided, the wallet must include them in the order provided as read-only, non-signer keys to the SystemProgram.Transfer or TokenProgram.Transfer/TokenProgram.TransferChecked instruction in the payment transaction. The values may or may not be unique to the payment request, and may or may not correspond to an account on Solana."

### Example
`solana:jupoGAk6QVHBtEdEUAsFhLRPCi5MK48bJZTu5AvBv6S?amount=100.00&spl-token=Gh9ZwEmdLJ8DscKNTkTqPbNwLNNBjuSzaG9Vp2KGtKJr&reference=2qjEcsqXQcdysoTpEgRkLJNJtFJCd8PnqmWoB8JqvPFx&label=mylabel&message=mymessage`

QR code:
https://chart.googleapis.com/chart?cht=qr&chs=250&chl=solana%3AjupoGAk6QVHBtEdEUAsFhLRPCi5MK48bJZTu5AvBv6S%3Famount%3D100.00%26spl-token%3DGh9ZwEmdLJ8DscKNTkTqPbNwLNNBjuSzaG9Vp2KGtKJr%26reference%3D2qjEcsqXQcdysoTpEgRkLJNJtFJCd8PnqmWoB8JqvPFx%26label%3Dmylabel%26message%3Dmymessage

Transaction on chain:
https://solscan.io/tx/2jdcYWmm1Uub9miqmerh5hrJ7YVWt471oeQHKpTiwfKmS7aM8ZYjErAmHLBrSLsPbopPrEY6iotiLCUj26U44NHf?cluster=devnet

When looking at the transaction details you can see that it includes the token address (Gh9ZwEmdLJ8DscKNTkTqPbNwLNNBjuSzaG9Vp2KGtKJr) in the accounts (SOL balance change), instead of the provided reference.

### Phantom Version
24.3.0

**Comment by adamdelphantom** [comment]
@matbech My apologies for the delayed response! I am investigating this issue.

**Reply by matbech** [reply]
@adamdelphantom I was wondering if you already had a chance to look at this?

**Reply by adamdelphantom** [reply]
@matbech Apologies again for delay in follow up! Thank you for your patience. From my investigations we added support for this field in August. I just reproduced using your provided QR code. Thank you for that!

So I understand better should the referenced account display any activity? Or is it just that the reference value showing up on the SOL balance change tab of solscan indicates the solana pay transaction is processed correctly?

I have a ticket with our engineering team for review and potential prioritization. I'll update you as I get more info on resolution and an eta. But feel free to check in on this thread anytime.

**Reply by matbech** [reply]
The amount is correct. It's just that the provided reference is not included in the transaction. Instead, the token address is included (instead of the reference).

**Reply by bhagyamudgal** [reply]
facing similar issue @adamdelphantom any update on this?

**Reply by adamdelphantom** [reply]
@bhagyamudgal and @matbech, apologies for the lack of support here. We are aware of the issue and continue to evaluate it for prioritization.

**Reply by adamdelphantom** [reply]
Once I can provide an eta I will post it here.

**Reply by bhagyamudgal** [reply]
> Once I can provide an eta I will post it here.

thanks, will be waiting for your reply

**Reply by illiaheria** [reply]
Is it fixed already?
